### ISSUE TITLE
RMarkdown italic+bold highlighting is incorrect

### ISSUE THREAD

**Issue by kohlkopf** [rstudio/rstudio#1699]
When one wishes to embolden and italicize a word or set of words in combination, the automatic highlighting malfunctions. All but the last asterisk become highlighted pink, indicative of being bold. Additionally, anything following this "hanging" asterisk is highlighted until another asterisk is reached.

In this example, the `b` indicates what is highlighted as bold, the `i` as what is highlighted as italic, and the `-` as what is not highlighted.

```
***The example*** highlights *this*
bbbbbb bbbbbbbbbb iiiiiiiiii i----- 
```
![image](https://user-images.githubusercontent.com/28633777/32260033-50e44ab0-be82-11e7-89c3-1d4c63fa6c38.png)

**Comment by dfalty** [comment]
Thanks @kohlkopf for the ticket. I can reproduce this on RSP 1.2.141-1 (RHEL 7.3) and Desktop 1.1.383 (Mac).

**Comment by kevinushey** [comment]
FWIW a workaround would be to combine underscores and stars, e.g.

```
_**Bold and Italic**_: <...>
```

**Comment by franknarf1** [comment]
Another example:

```
* Yada yada talking 'bout wildcards `attr_*` eh.
* Next bullet.
```

![italics](https://user-images.githubusercontent.com/4649190/37304162-38f69362-2607-11e8-833a-242731272ea6.png)

The first line is displayed as if italic, but it shouldn't interpret either of these as pertinent to italics:
* An asterisk for a bullet
* An asterisk inside code

**Comment by versipellis** [comment]
The workaround specified by @kevinushey no longer functions - the text renders italicized but not bolded.

**Comment by mine-cetinkaya-rundel** [comment]
Relatedly, 

- Multi word text that is italicized (with `*` or with `_`) is not being highlighted
- Non-styled text between two strings of text bolded with `__` on the same line is being highlighted 

I'm using 1.2.1030.

<img width="844" alt="screen shot 2018-10-07 at 9 05 04 am" src="https://user-images.githubusercontent.com/5965649/46582207-5be83700-ca11-11e8-95ef-687b8edd9ec7.png">

**Comment by kevinushey** [comment]
Thanks for spotting this @mine-cetinkaya-rundel -- we'll make sure to get this fixed up soon.

**Comment by dfalty** [comment]
Verified fixed on version 1.2.1090.
